### ISSUE TITLE
searchFunction deprecated

### ISSUE THREAD

**Issue by fogx** [typesense/showcase-hn-comments-semantic-search#1]
## Description

Since this repo serves as an example for the vector search implementation in the typesense adapter, its good to keep it updated or commented.
The used searchFunction is now deprecated, but i couldn't get the function to work with the recommended onStateChange. Would be nice to have an updated example or a note inside the code that it is ok to keep using searchFunction

## Steps to reproduce

go to this line inside an IDE
https://github.com/typesense/showcase-hn-comments-semantic-search/blob/8a33006cae58b425c53f56a64e1273e808cd9375/src/js/index.js#L101

## Expected Behavior

No errors

## Actual Behavior

'searchFunction' is deprecated.ts(6385)
@deprecated — use onStateChange instead

**Comment by tharropoulos** [comment]
The issue has been addressed on #2 

If you can, please consult the implementation there to verify that it works on your use-case as well!

You have to [deconstruct](https://github.com/tharropoulos/showcase-hn-comments-semantic-search/blob/4c4c975329e0e50ddb2dfc717c4e7a1dea92b339/src/js/index.js#L125) the `uiState` object in order to get the collection key you're searching against:
```javascript
    onStateChange({ uiState, setUiState }) {
      const { "hn-comments": state } = uiState;
      const query = state.query || "";
      const page = state.page || 0;
      const searchType = $("#search-type-select").val();

      const configure = {
        ...state.configure,
        typesenseVectorQuery:
          query && ["semantic", "hybrid"].includes(searchType)
            ? `embedding:([], k:200)`
            : null,
      };

      const newUiState = {
        ...uiState,
        "hn-comments": {
          ...state,
          configure,
          page,
        },
      };

      setUiState(newUiState);
```

**Comment by jasonbosco** [comment]
Resolved in #2
